### PR TITLE
feat: input sanitization for organization creation and editing

### LINKED ISSUE #451: Input Sanitization for Organization creation and editing
**Description:** 
Implement input sanitization and validation for all user-facing fields in the organization flows (create and edit). This includes organization name, description, contact email, phone number, website, social links, and address. Prevent invalid, malicious, or overly long inputs from being stored or processed.

**Acceptance Criteria:**
- [ ] All organization-related inputs are sanitized before creating or updating an organization
- [ ] Editing an organization cannot be saved with required fields emptied after sanitization
- [ ] Malicious inputs in any organization field are blocked with clear error messages
- [ ] Overly long values for name, description, contact details, and URLs are truncated or rejected according to defined limits.

### PR DESCRIPTION
**Pull request by walid-as**
# Description

## Purpose of the change
This PR implements **comprehensive input sanitization and validation** for all user-facing fields in the organization creation and editing flows. The solution prevents **injection attacks** (XSS, SQL, HTML), blocks **malicious inputs**, and enforces **reasonable length limits** while delivering a **consistent and secure user experience** aligned with our event form patterns.

**Key changes include:**
- **Major architectural refactor** of `OrganizationForm.kt` to unify its design, state management, and validation flow with the established event form architecture.
- **Robust validation logic** that makes name, description, email, and phone number mandatory and prevents saving if required fields become empty after sanitization.
- **Real-time user feedback** through character counters, inline warnings, and clear error messages upon submission.
- **Complete overhaul of the test suite** to ensure security, reliability, and behavioral consistency.

The implementation follows a defense-in-depth approach—sanitizing on input, validating before processing, and providing clear user feedback—while maintaining full backward compatibility.

## Related issues
Closes #451,  #459 and part of #452 

## How to test
* Check out the branch.  
* Run the app and navigate to organization creation or editing screens
* Verify that:
  - Character counters appear for title and description fields
  - Dangerous input patterns (HTML tags, SQL commands, script injections) are blocked
  - All organization fields (name, description, contact email, phone, website, social links, address) are properly sanitized
  - Overly long values are either truncated or rejected based on field-specific limits
  - Editing cannot be saved if required fields (name, description, email, phone) are empty after sanitization
  - The organization form’s behavior, feedback, and validation feel consistent with the event form
  - Existing functionality continues to work normally
 
* Run the updated test suites to confirm comprehensive coverage

<tr>
  <td align="center" style="padding: 20px;">
    <table align="center" cellpadding="0" cellspacing="0" style="margin: 0 auto;">
      <tr>
        <td align="center" style="text-align: center;">
          Screenshot from 2025-12-16 23-18-36
          <br>
          <em style="display: block; text-align: center; font-style: italic; margin-top: 10px;">Organization Form with Feedback for the User</em>
        </td>
      </tr>
    </table>
  </td>
</tr>

**Important Notes:**
- **Scope & Size:** ⚠️ This PR significantly exceeds the 500 LOC limit due to the **essential architectural refactor** required to unify form patterns and the addition/modification of **978 lines of comprehensive tests**. These tests are critical to validate the new sanitization logic, security protections, edge-case handling, and consistent UX across all organization fields.
- **User Experience:** Sanitization is applied as users type, providing immediate feedback. Validation errors are displayed upon form submission to avoid interrupting the user’s input flow.
- **LLM Assistance:** An LLM was used for the sole and exclusive purpose of improving the clarity and formatting of the PR description.

### CODE REVIEW

File: app/src/androidTest/java/ch/onepass/onepass/ui/organizer/EditOrganizationScreenTest.kt [hunk 1]
```diff
@@ -157,13 +174,189 @@ class EditOrganizationScreenTest {
...
+            OrganizationFormTestTags.COVER_IMAGE_BUTTON,
+            OrganizationFormTestTags.SUBMIT_BUTTON)
+
+    allFieldTags.forEach { tag ->
+      composeTestRule.onNodeWithTag(tag).performScrollTo().assertIsDisplayed()
+    }
+  }
+
+  @Test
+  fun editingPreservesExistingData() {
```
Comment (M-e-d-A) [inline app/src/androidTest/java/ch/onepass/onepass/ui/organizer/EditOrganizationScreenTest.kt]: This test doesn't actually verify that existing data is preserved—it just checks that the name field exists.

File: app/src/main/java/ch/onepass/onepass/ui/components/forms/FormFields.kt [hunk 2]
```diff
@@ -1,203 +1,96 @@
...
-    text: String,
+fun CompactFieldLabel(
+    label: String,
+    currentLength: Int,
+    maxLength: Int,
     modifier: Modifier = Modifier,
-    isLoading: Boolean = false,
-    enabled: Boolean = true
+    isError: Boolean = false,
+    testTag: String? = null
```
Comment (M-e-d-A) [inline app/src/main/java/ch/onepass/onepass/ui/components/forms/FormFields.kt]: The compact version looks good and matches the full version's behavior. Consistent API design.

Comment (walid-as) [reply app/src/main/java/ch/onepass/onepass/ui/components/forms/FormFields.kt]: Thanks.

File: app/src/main/java/ch/onepass/onepass/ui/organizer/OrganizationForm.kt [hunk 3]
```diff
@@ -126,79 +164,574 @@ fun OrganizerForm(
...
+ * Address input field with character counter.
+ *
+ * @param value Current address value
+ * @param onValueChange Callback when address changes
+ */
+@Composable
+private fun OrganizationAddressField(
+    value: String,
+    onValueChange: (String) -> Unit,
+) {
```
[the remark this answers is not in the document]
Comment (walid-as) [reply app/src/main/java/ch/onepass/onepass/ui/organizer/OrganizationForm.kt]: Thanks

File: app/src/main/java/ch/onepass/onepass/ui/organizer/OrganizationFormViewModel.kt [hunk 4]
```diff
@@ -173,7 +180,7 @@ class OrganizationFormViewModel(
    * @return An error message if invalid, null otherwise
    */
   private fun validateEmail(value: String): String? {
-    if (value.isBlank()) return null
+    if (value.isBlank()) return "Email is required"
     return if (!ValidationUtils.isValidEmail(value)) {
       "Invalid email"
     } else null
```
[the remark this answers is not in the document]
Comment (walid-as) [reply app/src/main/java/ch/onepass/onepass/ui/organizer/OrganizationFormViewModel.kt]: Yes it does, thanks

File: app/src/androidTest/java/ch/onepass/onepass/ui/organizer/CreateOrganizationScreenTest.kt [hunk 5]
```diff
@@ -152,33 +157,143 @@ class CreateOrganizationScreenTest {
...
+    // Enter some text
+    composeTestRule
+        .onNodeWithTag(OrganizationFormTestTags.NAME_FIELD)
+        .performScrollTo()
+        .performTextInput("Test")
+
+    // Character counter should show "4/50 characters" or similar
+    composeTestRule
+        .onNodeWithTag(OrganizationFormTestTags.NAME_CHAR_COUNT)
+        .assertTextContains("/", substring = true)
```
[the remark this answers is not in the document]
Comment (walid-as) [reply app/src/androidTest/java/ch/onepass/onepass/ui/organizer/CreateOrganizationScreenTest.kt]: Fixed.

**Comment by walid-as** [comment]
> Thanks for the thorough work on input sanitization and security improvements! I've reviewed the changes and left detailed comments throughout the code. Please review the inline feedback, along with a few general remarks below:
> 
> 
> 
> 
> Please feel free to push back on any of these points, or let me know if they are unrelated to your work and should therefore be addressed in a different context.

Thank you for the detailed testing feedback. I appreciate the thorough review, but these UI/UX and navigation issues fall outside the scope of this PR, which is focused specifically on **input sanitization and security improvements**. 

This allows each concern to be properly addressed in dedicated PRs without diluting the security enhancements at hand.
